src/cmd_all/src/single_node.rs

```diff
@@ -227,3 +268,254 @@ impl SingleNodeOpts {
+        let actual = format!("{:#?}", actual);
+        expect.assert_eq(&actual);
+    }
+
+    #[test]
+    fn test_parse_extra_opts() {
+        let raw_opts = "
+--meta-extra-opts=--advertise-addr 127.0.0.1:9999 --data-directory \"some path with spaces\" --state-store hummock+fs:///root/.risingwave/state_store
+--compute-extra-opts=--listen-addr 127.0.0.1:8888 --total-memory-bytes 123 --parallelism 10
+";
+        let actual = SingleNodeOpts::parse_from(raw_opts.lines());
+        check(
+            &actual,
+            expect![[r#"
+                SingleNodeOpts {
+                    prometheus_listener_addr: None,
+                    config_path: None,
+                    store_directory: None,
+                    meta_addr: None,
+                    compute_addr: None,
+                    frontend_addr: None,
+                    compactor_addr: None,
+                    meta_extra_opts: Some(
+                        "--advertise-addr 127.0.0.1:9999 --data-directory \"some path with spaces\" --state-store hummock+fs:///root/.risingwave/state_store",
+                    ),
+                    compute_extra_opts: Some(
+                        "--listen-addr 127.0.0.1:8888 --total-memory-bytes 123 --parallelism 10",
+                    ),
+                    frontend_extra_opts: None,
+                    compactor_extra_opts: None,
+                }"#]],
+        );
+        let actual_parsed = parse_single_node_opts(&actual).unwrap();
+        check(
+            actual_parsed,
+            expect![[r#"
+                ParsedSingleNodeOpts {
+                    meta_opts: Some(
+                        MetaNodeOpts {
+                            vpc_id: None,
+                            security_group_id: None,
+                            listen_addr: "127.0.0.1:5690",
+                            advertise_addr: "127.0.0.1:9999",
+                            dashboard_host: Some(
+                                "0.0.0.0:5691",
+                            ),
+                            prometheus_listener_addr: Some(
+                                "0.0.0.0:1250",
+                            ),
+                            etcd_endpoints: "",
+                            etcd_auth: false,
+                            etcd_username: "",
+                            etcd_password: [REDACTED alloc::string::String],
+                            sql_endpoint: Some(
+                                "sqlite:///Users/noelkwan/.risingwave/meta_store/single_node.db?mode=rwc",
+                            ),
+                            dashboard_ui_path: None,
+                            prometheus_endpoint: None,
+                            prometheus_selector: None,
+                            connector_rpc_endpoint: None,
+                            privatelink_endpoint_default_tags: None,
+                            config_path: "",
+                            backend: Some(
+                                Sql,
```

Comment:
The initial options we provided with `Sql` backend is preserved and not overridden from providing meta opts.